editoast/src/views/infra/auto_fixes.rs

```diff
@@ -433,4 +441,62 @@ mod test {
             .unwrap()
             .is_empty());
     }
+
+    #[rstest::rstest]
+    async fn invalid_switch_ports() {
+        let app = create_test_service().await;
+        let small_infra = small_infra(db_pool()).await;
+        let small_infra_id = small_infra.id();
+
+        let mut ports: HashMap<Identifier, TrackEndpoint> = HashMap::new();
+        ports.insert(
+            "WRONG".into(),
+            TrackEndpoint {
+                endpoint: Endpoint::End,
+                track: "TA1".into(),
+            },
+        );
+        ports.insert(
+            "B1".into(),
+            TrackEndpoint {
+                endpoint: Endpoint::Begin,
+                track: "TA3".into(),
+            },
+        );
+        ports.insert(
+            "B2".into(),
+            TrackEndpoint {
+                endpoint: Endpoint::Begin,
+                track: "TA4".into(),
+            },
+        );
```

Comment:
minor (style): for a bit shorter and to avoid `mut`, you can use `HashMap::from()`
```rust
                let ports = HashMap::from([
                    (
                        "WRONG".into(),
                        TrackEndpoint {
                            endpoint: Endpoint::End,
                            track: "TA1".into(),
                        },
                    ),
                    (
                        "B1".into(),
                        TrackEndpoint {
                            endpoint: Endpoint::Begin,
                            track: "TA3".into(),
                        },
                    ),
                    (
                        "B2".into(),
                        TrackEndpoint {
                            endpoint: Endpoint::Begin,
                            track: "TB4".into(),
                        },
                    ),
                ]);
```
(you can also do it right away in the Switch build)